src/core/send.c

```diff
@@ -1120,6 +1121,16 @@ QuicSendFlush(
         return TRUE;
     }
 
+    //
+    // Connection CID changes on idle state after an amount of time
+    //
+    if (Connection->Settings.DestCidUpdateIdleTimeoutMs != 0 &&
+        Send->LastFlushTimeValid &&
+        CxPlatTimeDiff64(Send->LastFlushTime, TimeNow) >= MS_TO_US(Connection->Settings.DestCidUpdateIdleTimeoutMs)) {
+        Connection->Stats.Misc.DestCidUpdateCount++;
```

Comment:
I know it might complicate your tests a tad, but I do think putting this in the function below would be a good idea.

Reply:
It shouldn't affect the test actually, since in test case it shouldn't retire all destination cids I think. I'm going to do that